syntax = "proto3";
package input;

message Spec {
  uint32 num_vertical_subdivisions = 1;
  uint32 matrix_width = 2;
  uint32 simdvector_bitlength = 3;
  enum integertype {
    uint8_t = 0;
    uint16_t = 1;
    uint32_t = 2;
    uint64_t = 3;   
    int8_t = 4;
    int16_t = 5;
    int32_t = 6;
    int64_t = 7;   
  };
  integertype integertype_to_use = 4;
}

message Column {
  repeated int64 elements = 1; 
//  repeated  google.protobuf.UInt64Value elements = 1; 
}

message Matrix {
  repeated Column columns = 1; 
}

message Root {
  Spec spec = 1; 
  repeated Matrix matrices = 2;
}
